datalad/interface/utils.py

```diff
@@ -107,6 +107,33 @@ def handle_dirty_dataset(ds, mode, msg=None):
         raise ValueError("unknown if-dirty mode '{}'".format(mode))
 
 
+def common_root_ds(paths):
+    """Return a common root dataset for a set of absolute paths
+
+    Note, that in opposition to get_tree_roots, this will figure a *common* root, not a set of roots.
+    In addition, this currently does actually resolve the paths rather than performing pure path operations.
+    """
+
+    from collections import Counter
```

Comment:
if coveralls is correct, this function is not tested at all -- could be a subject to a dedicated unittest (and a benchmark)